### DISCUSSION TITLE
How is path_shorten so fast  in `files`

### DISCUSSION THREAD

**Discussion by ArbitRandomUser** [ibhagwan/fzf-lua#1649] [General]
I implemented my own picker with fzf_live , running through a transform like this
```julia
fn_transform = function (x)
            return Path.shorten(fzflua.make_entry.file(x,fzfopts),3)
end
```
but this seems to be very laggy on large number of files.

however with `path_shorten=true` for `files` and running find files in my home folder fzflua does it in a breeze

**Comment by ArbitRandomUser** [comment]
here `Path=require"fzf-lua.path"`

**Comment by ArbitRandomUser** [comment]
ah alright , path_shorten can be passed to make_entry , 
i was doing this all wrong .

**Reply by ibhagwan** [reply]
> ah alright , path_shorten can be passed to make_entry , i was doing this all wrong .

Correct, it’s better to let the logic of make_entry handle that.

> I implemented my own picker with fzf_live , running through a transform like this

unfortunately using fn_transfom is running on the main thread, that’s due to the complexity of doing it on the external headless process, getting similar performance to the live_grep pickers is not straight forward and requires a bit more “hacking” (as the fn_transform) is loaded as a lua string by the external process.
